web/packages/teleport/src/Discover/Shared/ConfigureDiscoveryService/ConfigureDiscoveryService.tsx

```diff
@@ -0,0 +1,189 @@
+   * nextStep button
+   */
+  withCreateConfig?: boolean;
+}) {
+  const ctx = useTeleport();
+  const clusterId = ctx.storeUser.getClusterId();
+  const {
+    nextStep,
+    prevStep,
+    agentMeta,
+    updateAgentMeta,
+    resourceSpec,
+    emitErrorEvent,
+  } = useDiscover();
+
+  const {
+    attempt: createDiscoveryConfigAttempt,
+    setAttempt: setCreateDiscoveryConfigAttempt,
+  } = useAttempt('');
+
+  const [discoveryGroupName, setDiscoveryGroupName] = useState(
+    DEFAULT_DISCOVERY_GROUP_NON_CLOUD
+  );
+
+  const { storeUser } = useTeleport();
+
+  function handleNextStep() {
+    if (withCreateConfig) {
+      createDiscoveryCfg();
+      nextStep();
+      return;
+    }
+
+    updateAgentMeta({
+      ...agentMeta,
+      autoDiscovery: {
+        config: { name: '', aws: [], discoveryGroup: discoveryGroupName },
+      },
+    });
+    nextStep();
+  }
+
+  async function createDiscoveryCfg() {
+    try {
+      setCreateDiscoveryConfigAttempt({ status: 'processing' });
+      // Cloud has a discovery service automatically running so
+      // we have everything we need to create a
+      const discoveryConfig = await createDiscoveryConfig(clusterId, {
+        name: crypto.randomUUID(),
+        discoveryGroup: discoveryGroupName,
+        aws: [
+          {
+            types: ['rds'],
+            regions: [agentMeta.awsRegion],
+            tags: { 'vpc-id': [agentMeta.vpcId] },
+            integration: agentMeta.awsIntegration.name,
+          },
+        ],
+      });
+      setCreateDiscoveryConfigAttempt({ status: 'success' });
+      updateAgentMeta({
+        ...agentMeta,
+        autoDiscovery: {
+          config: discoveryConfig,
+        },
+      });
+    } catch (err) {
+      const message = getErrMessage(err);
+      setCreateDiscoveryConfigAttempt({
+        status: 'failed',
+        statusText: `failed to create discovery config: ${message}`,
+      });
+      emitErrorEvent(`failed to create discovery config: ${message}`);
+      return;
+    }
+  }
+
+  return (
+    <Box maxWidth="1000px">
+      <Header>Configure Teleport Discovery Service</Header>
+      {getResourceSpecificComp(resourceSpec.kind)}
+      <SelfHostedAutoDiscoverDirections
+        showSubHeader={false}
+        clusterPublicUrl={storeUser.state.cluster.publicURL}
+        discoveryGroupName={discoveryGroupName}
+        setDiscoveryGroupName={setDiscoveryGroupName}
+      />
+      <ActionButtons onProceed={handleNextStep} onPrev={prevStep} />
+      {createDiscoveryConfigAttempt.status !== '' && (
+        <CreatedDiscoveryConfigDialog
+          attempt={createDiscoveryConfigAttempt}
+          next={nextStep}
+          close={() => setCreateDiscoveryConfigAttempt({ status: '' })}
+          retry={handleNextStep}
+          region={agentMeta.awsRegion}
+          notifyAboutDelay={false}
+        />
+      )}
+    </Box>
+  );
+}
+
+function getResourceSpecificComp(kind: ResourceKind) {
```

Comment:
What is a "comp" in this context? Is it "component"? If so, perhaps we could use a more specific name here instead? Like "Warning" or "EnrollInfo".